### ISSUE TITLE
Feature request: expose URL and line number for ConsoleMessage object

### ISSUE THREAD

**Issue by stereobooster** [puppeteer/puppeteer#3029]
[LogEntry](https://chromedevtools.github.io/devtools-protocol/tot/Log#type-LogEntry) object provides url and lineNumber

```
url: string - URL of the resource if known.
lineNumber: integer - Line number in the resource.
```

can we also expose it?

Related source https://github.com/GoogleChrome/puppeteer/blob/25632133e2683c613c97d9088968c48e3c07a086/lib/Page.js#L167-L173

**Comment by aslushnikov** [comment]
@stereobooster sure, that's possible. Would you mind sharing a story why would you need this?

**Comment by stereobooster** [comment]
I see some console logs in the output, I want to track where they were called. It is not always trivial to do. https://github.com/stereobooster/react-snap is the tool I'm working on

**Comment by aslushnikov** [comment]
Reasonable.

**Comment by andrew-st-angelo-77media** [comment]
Would this be the same information given to the right of each console item, in DevTools, identifying the calling resource and the line number (eg. `application.min.js:13`)?  If so, that would be very useful.

**Comment by aslushnikov** [comment]
@andrew-st-angelo-77media yeah that would be it.

**Comment by silent-lad** [comment]
Can i work on this issue if no one is working on it ? @aslushnikov

**Comment by aslushnikov** [comment]
@silent-lad please go ahead

**Comment by tompere** [comment]
@stereobooster - I've created a pull request to support the above.

a few notes:
- I added `additionalInfo` method to ConsoleMessage under which `lineNumber` and `url` can be found. This is under the assumption that `additionalInfo` is a subset of [LogEntry optional params](https://chromedevtools.github.io/devtools-protocol/tot/Log#type-LogEntry), thus future requests for additional data can easily added.
- I could not reproduce a test (nor local) use-case where console message (`Log.entryAdded` event) is returned with actual `lineNumber`. Moreover, I could reproduce a violation log entry in devtools, with line number and url (e.g. invoking `document.write()`), yet the same page didn't returned that value when using puppeteer **or** a [3rd party client](https://github.com/cyrus-and/chrome-remote-interface) I used to work with RDP.
